### DISCUSSION TITLE
How to determine an image is dark or not enough brightness?

### DISCUSSION THREAD

**Discussion by dev-ptdi** [dlemstra/Magick.NET#699] [Help] [answered]
I use C# code below to determine an image is dark or not enough brightness, how to do this with Magick.Net?

                Bitmap bitmap = new Bitmap(_filename);

                var colors = new List<Color>();
                for (int x = 0; x < bitmap.Size.Width; x++)
                {
                    for (int y = 0; y < bitmap.Size.Height; y++)
                    {
                        colors.Add(bitmap.GetPixel(x, y));
                    }
                }

                float _calc_brightness = colors.Average(color => color.GetBrightness());

**Comment by dlemstra** [comment, marked as answer]
You could do that like this:

```C#
using (var image = new MagickImage(_filename))
{
    double _calc_brightness = image.Statistics().Composite().Mean;
}
```

**Reply by dev-ptdi** [reply]
Hi but the result is big different?

mine was : 0.12821941   but now I got 8273.42193287037

**Reply by dev-ptdi** [reply]
I need to determine an image is dark or blurry..

**Reply by dlemstra** [reply]
If you need a value between 0.0 and 1.0 you will need to use: `(1.0 / Quantum.Max) * _calc_brightness`. That will probably help you determine if the image is dark but I have no idea how you could use this to determine if an image is blurry.
